### PR TITLE
Tests for editing version details - part one

### PR DESCRIPTION
**Pull request by AlexandraMoga**
This PR includes:
- tests for editing an addon with the wrong user
- tests for editing version details
- tests for editing an addon license with valid and invalid values

### CODE REVIEW

File: tests/api/test_addon_uploads.py [hunk 1]
```diff
@@ -1752,6 +1752,160 @@ def test_upload_localized_extension_json_overwrite(base_url, session_auth):
...
+    addon = payloads.edit_addon_details['slug']
+    request = requests.get(
+        url=f'{base_url}{_addon_create}{addon}',
+        headers={'Authorization': f'Session {session_auth}'},
+    )
+    # get the version id of the version we want to edit
+    version = request.json()['current_version']['id']
+    # set a few unaccepted/invalid data types as license slugs
+    invalid_license = ['', None, False, 123, 'random slug', {'value': 'some value'}]
+    for slug in invalid_license:
```
Comment (jrbenny35) [inline tests/api/test_addon_uploads.py:1849]: Any reason you didn't parametrize this?

**Review by jrbenny35 — CHANGES_REQUESTED** [review]
One question

File: tests/api/test_addon_uploads.py (hunk 1 above)
Comment (AlexandraMoga) [reply tests/api/test_addon_uploads.py:1849]: I think I didn't want to repeat the version id request for each case, but that shouldn't be too taxing on the api. I think I could switch to a parameterize form.

File: tests/api/test_addon_uploads.py [hunk 2]
```diff
@@ -1833,9 +1833,28 @@ def test_edit_version_set_custom_license(base_url, session_auth):
...
+    ],
+    ids=[
+        'Empty string',
+        'None/No value',
+        'Boolean',
+        'Integer',
+        'Random string',
+        'Dictionary',
+    ],
+)
```
Comment (jrbenny35) [inline tests/api/test_addon_uploads.py:1854]: Love it!

**Review by jrbenny35 — APPROVED** [review]
r+, sorry for the late review

**Pull request merged**